runtime/interp.c

```diff
@@ -210,11 +210,18 @@ Caml_inline void check_trap_barrier_for_effect
 #define SP_REG asm("%r17")
 #define ACCU_REG asm("%r16")
 #endif
-#ifdef __mc68000__
+#ifdef __mc68000__ /* { */
+#ifdef __linux__
```

Comment:
Should this rather test for the libc? If it's glibc-specific we could try `__GNU_LIBRARY__` or `__GLIBC__` or `__GLIBC_MINOR__`. What does e.g., musl do?

Reply:
I considered this, but there is always the risk of mixing objects compiled against glibc with objects compiled against musl (which does not reserve `%a5`), so I opted for a more conservative check.